crates/types/src/identifiers.rs

```diff
@@ -615,7 +615,7 @@ impl IdempotencyId {
         idempotency_key: ByteString,
     ) -> Self {
         IdempotencyId {
-            service_name: invocation_target.handler_name().clone(),
+            service_name: invocation_target.service_name().clone(),
```

Comment:
What's the impact on existing Restate deployments that have idempotency keys stored? Will previously idempotent invocations be re-executed because we changed the `service_name` value? If there is no way to prevent similar effects from happening, we certainly need to list this as part of the release notes. It would also be great to include a description of the side effects in the commit message for documentation purposes.